wade-review.md

```diff
@@ -0,0 +1,596 @@
+## Getting started with Kafka Streams: The Processing Process
+
+
+Now, I'll pivot to building a Kafka Streams application that processes the 'github-pull_requests' events, creating an up-to-date ratio of open/closed pull requests.
+
+If you want to follow along with this part, check out the GitHub repo in the links below. I’m going to focus on the domain logic given our timeframe, so if you really want to dive in, I suggest consulting the codebase. 
+
+
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> {
+}
+```
+
+Now I'll  write the ProcessorSupplier, using the class provided by KafkaStreams. I'll use the record shape supplied by GitHubPRInfo as well as the object provided by GitHubPrStateCounter in our models folder.
+
+---
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+}
+```
+
+Our store key will be named 'state_counter'. 
+
+---
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
+	}
+}
+```
+
+Now, let's create a new processor instance to return. 
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
+
+    private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+}
+
+```
+
+Then, I’ll initialize it with our state store. The init method schedules a punctuation to fire every second, printing the ratio in the store. 
```

Comment:
There's a lot of questions for me in this area.

1. What is a state store? What purpose does it serve?
2. What is a punctuation? When I hear that word I think of commas and periods, but this is obviously something different.
3. I see that we obtain a value from the store, then print it. But what is the "context.forward" bit about?

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/4d3dc27b380196a10fc255173f804586b928bb83